wpilibc/src/main/native/cpp/trajectory/constraint/DifferentialDriveVoltageConstraint.cpp

```diff
@@ -24,7 +22,11 @@ DifferentialDriveVoltageConstraint::DifferentialDriveVoltageConstraint(
 units::meters_per_second_t DifferentialDriveVoltageConstraint::MaxVelocity(
     const Pose2d& pose, curvature_t curvature,
     units::meters_per_second_t velocity) {
-  return units::meters_per_second_t(std::numeric_limits<double>::max());
```

Comment:
Doesn't this change the behavior of the constraint? In essence, this is now doing what the kinematics constraint did, and changes the behavior of this constraint (because it did not limit velocity before).

Reply:
DifferentialDriveVoltageConstraint violates constraints if you don't have this. The feedforward voltage exceeds the max voltage specified by the constraint if you don't limit the chassis velocity to keep the wheel velocities below it. Adding the diff drive kinematics constraint separately fixes this, but that coupling is invisible and thus DifferentialDriveVoltageConstraint is a footgun.

Reply:
After further testing, using a diff drive kinematics constraint still results in constraint violations. Capping the wheel velocities to the unconstrained chassis velocity works though (which is what this does). I'm not sure why yet.